src/script/cpp_api/s_base.cpp

```diff
@@ -121,13 +121,20 @@ bool ScriptApiBase::loadMod(const std::string &scriptpath,
 		errorstream<<"Error loading mod \""<<modname
 				<<"\": modname does not follow naming conventions: "
 				<<"Only chararacters [a-z0-9_] are allowed."<<std::endl;
+
+		if (error) {
+			(*error)  = "Error loading mod ";
+			(*error) += scriptpath;
+			(*error) += ": modname does not follow naming conventions: ";
+			(*error) += "Only chararacters [a-z0-9_] are allowed.";
+		}
```

Comment:
This is essentially duplicate code, find a way to only create the error message once.